sdk/src/types/block/semantic.rs

```diff
@@ -368,11 +435,31 @@ pub fn semantic_validation(
         return Ok(Some(TransactionFailureReason::SumInputsOutputsAmountMismatch));
     }
 
+    if context.input_mana > context.output_mana
+        && context
+            .essence
+            .capabilities()
+            .has_capability(TransactionCapabilityFlag::BurnMana)
+    {
+        // TODO: better failure reason incoming?
+        return Ok(Some(TransactionFailureReason::SemanticValidationFailed));
+    }
+
     let mut native_token_ids = HashSet::new();
 
     // Validation of input native tokens.
     for (token_id, _input_amount) in context.input_native_tokens.iter() {
         native_token_ids.insert(token_id);
+
+        if !context.output_native_tokens.contains_key(token_id)
```

Comment:
It's more complex than that, you could have NTs in outputs and still burn some. Have a look at `FoundryOutput::transition_inner`, we are able to identify more precisely what is being burned.